### PR TITLE
[monorepo] Upgrade to Node12

### LINKED ISSUE #1801: Upgrade to Node 12
Node 10 is moving to maintenance mode very soon on 2020-05-19.

### PR DESCRIPTION
**Pull request by geoknee**
Closes #1801 
Some of the problems / fixes / sacrifices I came across

Necessary (ish)
---

- Node 12.16 worked but Node 12.18 did not work.

- Duplicated react types:
https://github.com/DefinitelyTyped/DefinitelyTyped/issues/33822

- ENOBUFS with etherlime/ native solc compilation (this had to be turned OFF): 
ethereum/solidity#8133

- various missing dependencies were fixed (monorepo can obscure this problem)

- we get an error during `yarn install` `warning Error running install script for optional dependency: "/home/circleci/project/node_modules/scrypt: Command failed`. It seems mostly inconsequential apart from a lot of noise in the console explaining the error.
 

Not necessary, but nice
---
- web3torrent now uses ethers, not web3js (we actually use only 1 tiny function)
- etherlime upgrade

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -7,23 +7,23 @@ orbs:
 
 defaults: &defaults
   environment:
-    USE_NATIVE_SOLC: true
+    USE_NATIVE_SOLC: false
   working_directory: /home/circleci/project
   resource_class: medium
   docker:
-    - image: counterfactual/statechannels:0.6.9 # Fast contract compilation with solc installed
+    - image: geoknee/statechannels:node12.16-solc0.6.2 # Fast contract compilation with solc installed
```
Comment (geoknee) [inline .circleci/config.yml:14]: TODO use a `counterfactual/` name-spaced image

**Comment by snario** [comment]
Might want to ensure `.nvmrc` and any `engine` fields in `package.json` fields are updated.

**Comment by geoknee** [comment]
> Might want to ensure `.nvmrc` and any `engine` fields in `package.json` fields are updated.

I think I got all the engines. `.nvmrc` in fa243ac

**Comment by snario** [comment]
Checked out the repo and:

- Ran `nvm i` locally to get Node 12.16.3
- Ran `yarn clean` to remove the old `node_modules` etc
- Ran `yarn` to install everything back
- I had to set `USE_NATIVE_SOLC=false` because `true` no longer works :(
- Ran all `e2e-tests` with `web3torrent` or `optional` regex and all seemed to work:

```
Test Suites: 5 passed, 5 total
Tests:       5 passed, 5 total
Snapshots:   0 total
Time:        244.585s
Ran all test suites matching /web3torrent|optional/i.
✨  Done in 248.24s.
```

So, LGTM.

**Pull request merged**